File: backend/app/crawler/selectors.py
"""Configurable selectors and page-state hints for visible RedNote/Xiaohongshu import.

Both sites can change markup often. Treat these selectors as editable hints,
not as a guarantee. The extractor also falls back to URL/text heuristics so
import does not depend on a single fragile selector.
"""

CARD_SELECTORS = [
    "section.note-item",
    "div.note-item",
    "div.feeds-page div.note-item",
    "a[href*='/explore/']",
    "a[href*='/discovery/item/']",
    "a[href*='/search_result/']",
]

XIAOHONGSHU_CARD_SELECTORS = CARD_SELECTORS

REDNOTE_CARD_SELECTORS = [
    "section.note-item",
    "div.note-item",
    "div[class*='note']",
    "section[class*='note']",
    "div[class*='card']",
    "a[href*='/explore/']",
    "a[href*='/discovery/item/']",
    "a[href*='/note/']",
    "a[href*='/item/']",
]

AUTHOR_SELECTORS = [
    ".author",
    ".name",
    ".user-name",
    "[class*='author']",
    "[class*='name']",
]

TITLE_SELECTORS = [
    ".title",
    ".note-title",
    "[class*='title']",
]

LINK_PATH_HINTS = [
    "/explore/",
    "/discovery/item/",
    "/search_result/",
    "/item/",
]

XIAOHONGSHU_LINK_PATH_HINTS = LINK_PATH_HINTS

REDNOTE_LINK_PATH_HINTS = [
    "/explore/",
    "/discovery/item/",
    "/search_result/",
    "/note/",
    "/notes/",
    "/item/",
    "/post/",
]

SITE_CARD_SELECTORS = {
    "rednote": REDNOTE_CARD_SELECTORS,
    "xiaohongshu": XIAOHONGSHU_CARD_SELECTORS,
}

SITE_LINK_PATH_HINTS = {
    "rednote": REDNOTE_LINK_PATH_HINTS,
    "xiaohongshu": XIAOHONGSHU_LINK_PATH_HINTS,
}

LOGIN_OR_CHALLENGE_URL_HINTS = [
    "login",
    "signin",
    "passport",
    "captcha",
    "verify",
    "security",
]

LOGIN_TEXT_HINTS = [
    "登录",
    "扫码登录",
    "手机号登录",
    "密码登录",
    "login",
    "sign in",
]

CHALLENGE_TEXT_HINTS = [
    "验证码",
    "安全验证",
    "滑动验证",
    "请完成验证",
    "captcha",
    "verify",
    "verification",
    "security check",
]

LOGIN_OR_CHALLENGE_TEXT_HINTS = [
    *LOGIN_TEXT_HINTS,
    *CHALLENGE_TEXT_HINTS,
]

AUTHENTICATED_TEXT_HINTS = [
    "消息",
    "通知",
    "发布",
    "创作中心",
    "我的",
    "退出登录",
    "个人主页",
]

AUTHENTICATED_SELECTOR_HINTS = [
    "a[href*='/user/profile/']",
    "[class*='avatar']",
    "[class*='user']",
    "[class*='profile']",
]

File: extension/src/lib/selectors.js
// Selector / URL hints ported from backend/app/crawler/selectors.py.
export const REDNOTE_CARD_SELECTORS = [
  "section.note-item", "div.note-item", "div[class*='note']", "section[class*='note']",
  "div[class*='card']", "a[href*='/explore/']", "a[href*='/discovery/item/']",
  "a[href*='/note/']", "a[href*='/item/']",
];
export const XIAOHONGSHU_CARD_SELECTORS = [
  "section.note-item", "div.note-item", "div.feeds-page div.note-item",
  "a[href*='/explore/']", "a[href*='/discovery/item/']", "a[href*='/search_result/']",
];
export const SITE_CARD_SELECTORS = { rednote: REDNOTE_CARD_SELECTORS, xiaohongshu: XIAOHONGSHU_CARD_SELECTORS };

export const TITLE_SELECTORS = [".title", ".note-title", "[class*='title']"];
export const AUTHOR_SELECTORS = [".author", ".name", ".user-name", "[class*='author']", "[class*='name']"];

export const REDNOTE_LINK_PATH_HINTS = ["/explore/", "/discovery/item/", "/search_result/", "/note/", "/notes/", "/item/", "/post/"];
export const XIAOHONGSHU_LINK_PATH_HINTS = ["/explore/", "/discovery/item/", "/search_result/", "/item/"];
export const SITE_LINK_PATH_HINTS = { rednote: REDNOTE_LINK_PATH_HINTS, xiaohongshu: XIAOHONGSHU_LINK_PATH_HINTS };

export const LOGIN_TEXT_HINTS = ["登录", "扫码登录", "手机号登录", "密码登录", "login", "sign in"];
export const CHALLENGE_TEXT_HINTS = ["验证码", "安全验证", "滑动验证", "请完成验证", "captcha", "verify", "verification", "security check"];
export const AUTHENTICATED_TEXT_HINTS = ["消息", "通知", "发布", "创作中心", "我的", "退出登录", "个人主页"];

export const SITE_MODES = {
  rednote: { site_key: "rednote", display_name: "RedNote", base_url: "https://www.rednote.com", explore_url: "https://www.rednote.com/explore", domains: ["www.rednote.com", "rednote.com"] },
  xiaohongshu: { site_key: "xiaohongshu", display_name: "Xiaohongshu", base_url: "https://www.xiaohongshu.com", explore_url: "https://www.xiaohongshu.com/explore", domains: ["www.xiaohongshu.com", "xiaohongshu.com"] },
};
